### PR TITLE
Internal: [Grid] Internal: Layout issues in Grid [ED-10201]

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: assets/dev/scss/frontend/_container.scss [hunk 1]
```diff
@@ -30,7 +30,8 @@
 	--position: relative;
 	--z-index: revert;
 	--overflow: visible;
-	--gap: var( --widgets-spacing, 20px ); // Use the gap from `Site Settings` or default to 20px.
+	--column-gap: var( --widgets-spacing, 20px ); // Use the gap from `Site Settings` or default to 20px.
+	--row-gap: var( --widgets-spacing, 20px ); // Use the gap from `Site Settings` or default to 20px.
```
Comment (hein-obox) [inline assets/dev/scss/frontend/_container.scss:34]: The combination of having two gap values inside the --row seemed to cause problems. Therefore I split up the values for the column and the row gap.

File: includes/controls/groups/grid-container.php [hunk 2]
```diff
@@ -149,19 +149,18 @@ protected function init_fields() {
 			'label' => esc_html_x( 'Justify Content', 'Grid Container Control', 'elementor' ),
 			'type' => Controls_Manager::CHOOSE,
 			'label_block' => true,
-			'default' => '',
 			'options' => [
 				'start' => [
 					'title' => esc_html_x( 'Start', 'Grid Container Control', 'elementor' ),
-					'icon' => 'eicon-flex eicon-justify-start-h',
+					'icon' => 'eicon-flex eicon-justify-start-v',
```
Comment (hein-obox) [inline includes/controls/groups/grid-container.php:155]: Unrelated fix. 
The horizontal and vertical icons were swopped.

Comment (davseve) [reply includes/controls/groups/grid-container.php:155]: Please remove all the icon fix related code from this PR.
Aviv will ask Louis to add it to [his PR ](https://elementor.atlassian.net/browse/ED-9891)

**Review by davseve — CHANGES_REQUESTED** [review]

File: assets/dev/scss/frontend/_container.scss (hunk 1 above)
Comment (davseve) [reply assets/dev/scss/frontend/_container.scss:34]: Shouldn't removing these properties in  `assets/dev/scss/frontend/_global.scss` be better?
The problem is here.
```
		column-gap: var( --grid-column-gap, var( --gap ) );
		row-gap: var( --grid-row-gap, var( --gap ) );
```

**Review by davseve — CHANGES_REQUESTED** [review]

File: assets/dev/scss/frontend/_container.scss (hunk 1 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/_container.scss]: Fixed.

File: includes/controls/groups/grid-container.php (hunk 2 above)
Comment (hein-obox) [reply includes/controls/groups/grid-container.php]: Updated

**Review by davseve — APPROVED** [review]

File: assets/dev/scss/frontend/_container.scss [hunk 3]
```diff
@@ -101,8 +101,8 @@
 
 	&-boxed.e-grid {
 		justify-items: initial;
-		grid-template-columns: initial;
-		grid-template-rows: initial;
+		grid-template-columns: 1fr;
```
Comment (louiswol94) [inline assets/dev/scss/frontend/_container.scss:104]:
```suggestion
		grid-template-columns: repeat(3, 1fr);
```
Just making sure? This is the default according to the spec.

File: assets/dev/scss/frontend/_container.scss [hunk 4]
```diff
@@ -101,8 +101,8 @@
...
+		grid-template-rows: 1fr;
```
Comment (louiswol94) [inline assets/dev/scss/frontend/_container.scss:105]:
```suggestion
		grid-template-rows: repeat(2, 1fr);
```
Just making sure? This is the default according to the spec.

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: assets/dev/scss/frontend/_container.scss (hunk 3 above)
Comment (louiswol94) [reply assets/dev/scss/frontend/_container.scss:104]: @hein-obox We also have a variable for these in the PHP? `--e-con-grid-template-rows` and `--e-con-grid-template-columns`

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: assets/dev/scss/frontend/_container.scss (hunk 3 above)
Comment (hein-obox) [reply assets/dev/scss/frontend/_container.scss:104]: @louiswol94 

My change only applies to the wrapper container (&-boxed.e-grid), in case there is an inner container. The wrapper needs to stay full width. This doesn't apply to the display of the content inside the container itself.

I hope I am making sense.

**Review by udidol — APPROVED** [review]

**Review by louiswol94 — APPROVED** [review]
